staging/src/k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap/default.go

```diff
@@ -82,6 +82,12 @@ var (
 	}
 )
 
+func makeInt32Ptr(v int32) *int32 {
+	p := new(int32)
+	*p = v
+	return p
+}
+
 // Mandatory PriorityLevelConfiguration objects
 var (
 	MandatoryPriorityLevelConfigurationExempt = newPriorityLevelConfiguration(
```

Comment:
I'm curious what prompted updating these, if there was a test that noticed/failed when these didn't match their round-tripped/defaulted versions?

Reply:
yes, there is a case here:
https://github.com/kubernetes/kubernetes/blob/4d50d28777795b01195368b27964458d798248dd/pkg/apis/flowcontrol/validation/validation.go#L348-L356

when a new mandatory `PriorityLevelConfiguration` object is created, the validation logic ensures that the spec defined in the source code (not defaulted before this PR) matches the object being created.

The integration job was failing due to this issue. Do you think deep copying and defaulting the spec (in the source file) before the comparison is a more preferable approach here?

currently, we have a test that verifies that defaulting any bootstrap object in the source code does not change the spec:
https://github.com/kubernetes/kubernetes/pull/112830/commits/69f3626d74470c9bc80895af8f895046ed866960#diff-91b89583335fd122fb47dc639206dcace01bf7812d7db0ea7eca697b17b29e50R31-R59

Reply:
> currently, we also have a test that verifies that defaulting bootstrap object(s) in the source code does not change the spec

That was what I was expecting. The check in API validation was surprising to me. I need to think about whether making API validation compare the object in the request to the compiled-in object introduces any issues with skewed servers. That doesn't block this PR though.